Docs to guide migrating users to new design of handling failed messages (#45)
upstash/docs: qstash/migrating/v2.mdx

Instead of creating schedules through the `/v2/publish` endpoint, there is a dedicated `/v2/schedules` api now.

### Where are the messages in V2 ?

A big difference on v2 is that we no longer keep the messages to be accessed via [Get Message](/qstash/api/messages/get) to scale better.

> - a message is removed from the messages list if the call was successfull.

Reviewer: ```suggestion
- A message is removed from the messages list if the call was successful.
```

> - If the pubhlish is failed, it will be moved to [DLQs](/qstash/features/dlq) when it is failed.

Reviewer: ```suggestion
- If the publish is failed, it will be moved to [DLQs](/qstash/features/dlq).
```

- This endpoint will return a message only if a message is retrying. 

> A common workflow was to check the message via message id [Get Message](/qstash/api/messages/get) to see if it was successfull or not.

Reviewer: ```suggestion
A common workflow was to check the message via message id [Get Message](/qstash/api/messages/get) to see if it was successful or not.
```

This was not a good workflow because, it is not clear when to check this endpoint to take an action. 

> Instead v2, to check a message is failed and act on it, you can:

Reviewer: ```suggestion
Instead in v2, to check a message is failed and act on it, you can:
```

1. Use [Upstash-Failure-Callback](/qstash/features/callbacks#what-is-a-failure-callback). Give the `Upstash-Failure-Callback` when publishing 
the message to be called when the message is failed to be delivered. In the given endpoint:

    - Handle the message. 
    - Remove it from DLQ using [Delete a Message From DLQ](/qstash/api/dlq/deleteMessage) via the `dlqId` delivered to [Upstash-Failure-Callback](/qstash/features/callbacks#what-is-a-failure-callback).

> 2. Use [DLQs](/qstash/features/dlq). This means that you need to a continously running task to read the messages from DLQ. 

Reviewer: ```suggestion
2. Use [DLQs](/qstash/features/dlq). This means that you need a continuously running task to read the messages from DLQ. 
```

> The workflow in this case look like this:

Reviewer: ```suggestion
The workflow in this case looks like this:
```
